R/github-labels.R

```diff
@@ -1,22 +1,41 @@
-#' Create standard github labels for labelling issues.
+#' Manage GitHub issue labels.
 #'
-#' This function creates new labels and changes colours (if needed).
-#' It does not generally remove labels, unless you explicit ask to
-#' remove GitHub's defaults.
+#' @description `use_github_labels()` creates new labels and/or changes label
+#' colours. It does not generally remove labels, unless you explicitly ask to
+#' remove the default GitHub labels that are not present in the labels you
+#' provide via `labels`.
 #'
-#' @param delete_default If `TRUE`, will remove default labels.
+#' `gh_labels()` returns the labels and colours commonly used by tidyverse
+#' packages.
+#'
+#' @param labels Named character vector of labels. The names are the label text,
+#'   such as "bug", and the values are the label colours in hexadecimal, such as
+#'   "d73a4a". First, labels that don't yet exist are created, then label
+#'   colours are updated.
+#' @param delete_default If `TRUE`, will remove GitHub default labels that do
+#'   not appear in the `labels` vector (presumably defaults that aren't relevant
+#'   to your workflow).
 #' @inheritParams use_github_links
+#' @name use_github_labels
+NULL
+
+#' @rdname use_github_labels
 #' @export
-use_github_labels <- function(delete_default = FALSE,
+#' @examples
+#' \dontrun{
+#' ## typical use in, e.g., a new tidyverse project
+#' use_github_labels(delete_default = TRUE)
+#' }
+use_github_labels <- function(labels = gh_labels(),
+                              delete_default = FALSE,
```

Comment:
Should this be `remove = FALSE`?

Reply:
Are you suggesting that `remove = TRUE` requests to drop labels that are not present in `labels`, regardless of whether they are GitHub defaults?

Based on my interactions with usethis labels while working on this, that would create more work. For example, pretty sure I would have accidentally deleted the [non-package label](https://github.com/r-lib/usethis/labels/non-package) by now.